spec.html

```diff
@@ -344,115 +269,165 @@ <h1>
+      <emu-clause id="sec-asyncsnapshot.prototype.run">
+        <h1>AsyncSnapshot.prototype.run ( _func_, ..._args_ )</h1>
+        <p>This method performs the following steps when called:</p>
+        <emu-alg>
+          1. Let _asyncSnapshot_ be the *this* value.
+          1. Perform ? RequireInternalSlot(_asyncSnapshot_, [[AsyncSnapshotMapping]]).
+          1. Let _snapshotMapping_ be _asyncSnapshot_.[[AsyncSnapshotMapping]].
+          1. Let _previousContextMapping_ be AsyncContextSnapshot().
+          1. AsyncContextSwap(_snapshotMapping_).
+          1. Let _result_ be Completion(Call(_func_, *undefined*, _args_)).
+          1. AsyncContextSwap(_previousContextMapping_).
+          1. Return _result_.
+        </emu-alg>
+      </emu-clause>
+    </emu-clause>
+
+    <emu-clause id="sec-properties-of-asyncsnapshot-instances">
+      <h1>Properties of AsyncSnapshot Instances</h1>
+      <p>AsyncSnapshot instances are ordinary objects that inherit properties from the AsyncSnapshot prototype object (the intrinsic, %AsyncSnapshot.prototype%). AsyncSnapshot instances are initially created with the internal slots described in <emu-xref href="#table-internal-slots-of-asyncsnapshot-instances"></emu-xref>.</p>
+
+      <emu-table id="table-internal-slots-of-asyncsnapshot-instances" caption="Internal Slots of AsyncSnapshot Instances">
+        <table>
+          <tr>
+            <th>
+              Internal Slot
+            </th>
+            <th>
+              Type
+            </th>
+            <th>
+              Description
+            </th>
+          </tr>
+          <tr>
+            <td>
+              [[AsyncSnapshotMapping]]
+            </td>
+            <td>
+              a List of Async Context Mapping Records
+            </td>
+            <td>
+              Represents the snapshotted surrounding agent's Agent Record's [[AsyncContextMapping]] of the AsyncSnapshot instance.
+            </td>
+          </tr>
+        </table>
+      </emu-table>
+    </emu-clause>
+  </emu-clause>
+
+  <emu-clause id="sec-asynclocal-objects">
+    <h1>AsyncLocal Objects</h1>
+
+    <emu-clause id="sec-asynclocal-constructor">
+      <h1>The AsyncLocal Constructor</h1>
+      <p>The AsyncLocal constructor:</p>
+      <ul>
+        <li>is <dfn>%AsyncLocal%</dfn>.</li>
+        <li>is the initial value of the *"AsyncLocal"* property of the global object.</li>
+        <li>creates and initializes a new AsyncLocal when called as a constructor.</li>
+        <li>is not intended to be called as a function and will throw an exception when called in that manner.</li>
+        <li>may be used as the value in an `extends` clause of a class definition. Subclass constructors that intend to inherit the specified AsyncLocal behaviour must include a `super` call to the AsyncLocal constructor to create and initialize the subclass instance with the internal state necessary to support the `AsyncLocal` and `AsyncLocal.prototype` built-in methods.</li>
```

Comment:
```suggestion
        <li>may be used as the value in an `extends` clause of a class definition. Subclass constructors that intend to inherit the specified AsyncLocal behaviour must include a `super` call to the AsyncLocal constructor to create and initialize the subclass instance with the internal state necessary to support the `AsyncLocal.prototype` built-in methods.</li>
```

The `AsyncLocal` constructor no longer has built-in methods. Same for the `AsyncSnapshot` constructor.